### PR TITLE
Tests - tweaks

### PR DESCRIPTION
**Pull request by SpacePossum**
Clean ups and such, little tree shake from other PR's and stashes.

### CODE REVIEW

File: tests/Console/Command/ReadmeCommandTest.php [hunk 1]
```diff
@@ -28,20 +29,33 @@
 {
     public function testIfReadmeFileIsCorrect()
     {
-        if (!class_exists('Symfony\Component\Console\Output\BufferedOutput')) {
-            $this->markTestSkipped('Unsupported symfony/console version, Symfony\Component\Console\Output\BufferedOutput was added in 2.4.');
-        }
+        $readmeFile = __DIR__.'/../../../README.rst';
+        $this->assertFileExists($readmeFile, sprintf('README file "%s" not found.', $readmeFile)); // switch to `assertFileIsReadable` on PHPUnit6
+        $this->assertTrue(is_readable($readmeFile), sprintf('Cannot not read "%s".', $readmeFile));
```
Comment (julienfalque) [inline tests/Console/Command/ReadmeCommandTest.php]: Double negation.

File: tests/Console/Command/ReadmeCommandTest.php [hunk 2]
```diff
@@ -28,20 +29,33 @@
...
+        $this->assertTrue(is_file($readmeFile), sprintf('Expected file "%s".', $readmeFile));
```
Comment (julienfalque) [inline tests/Console/Command/ReadmeCommandTest.php]: _Expected "%s" to be a file_?

File: tests/Console/Command/ReadmeCommandTest.php [hunk 3]
```diff
@@ -28,20 +29,33 @@
...
+        $fileContent = file_get_contents($readmeFile);
+        $this->assertInternalType('string', $fileContent, sprintf('Failed to get content of "%s"', $readmeFile));
```
Comment (julienfalque) [inline tests/Console/Command/ReadmeCommandTest.php]: Not sure about doing an actual assertion on this as it's a test implementation detail. I would use `fail()` or throw an exception instead.

File: tests/Console/Command/ReadmeCommandTest.php [hunk 4]
```diff
@@ -28,20 +29,33 @@
...
-        $fileContent = file_get_contents(__DIR__.'/../../../README.rst');
+        $output->setVerbosity(OutputInterface::VERBOSITY_VERY_VERBOSE);
+        $output->setDecorated(false);
+
+        $exitCode = $app->get('readme')->run($input, $output);
+        $output = $output->fetch();
+        // normalize line breaks, these are not important for the tests
+        $output = str_replace(PHP_EOL, "\n", $output);
+
+        $this->assertSame(
```
Comment (julienfalque) [inline tests/Console/Command/ReadmeCommandTest.php]: This assertion can be done before fetching output, which would remove the need to use a variable for the exit code.

File: tests/Console/Command/ReadmeCommandTest.php [hunk 5]
```diff
@@ -28,20 +29,33 @@
...
+            0,
+            $exitCode,
+            sprintf("readme command did not exit successful.\n%s", $output)
```
Comment (julienfalque) [inline tests/Console/Command/ReadmeCommandTest.php]: Typo: _successful**ly**_.

File: tests/Console/Command/ReadmeCommandTest.php (hunk 1 above)
Comment (SpacePossum) [inline tests/Console/Command/ReadmeCommandTest.php]: not sure I understand

File: tests/Console/Command/ReadmeCommandTest.php (hunk 3 above)
Comment (SpacePossum) [inline tests/Console/Command/ReadmeCommandTest.php]: implementation detail of PHo, i. e. string vs false seems ligit to me

Comment (SpacePossum) [inline tests/Console/Command/ReadmeCommandTest.php]: PHP*

File: tests/Console/Command/ReadmeCommandTest.php (hunk 2 above)
Comment (SpacePossum) [inline tests/Console/Command/ReadmeCommandTest.php]: would it differ here?

File: tests/Console/Command/ReadmeCommandTest.php (hunk 4 above)
Comment (SpacePossum) [inline tests/Console/Command/ReadmeCommandTest.php]: it cannot, as the output can only be fetched one time the output object

File: tests/Console/Command/ReadmeCommandTest.php (hunk 5 above)
Comment (SpacePossum) [inline tests/Console/Command/ReadmeCommandTest.php]: not sure, both forms seems to be allowed following google

File: tests/Console/Command/ReadmeCommandTest.php (hunk 1 above)
Comment (keradus) [inline tests/Console/Command/ReadmeCommandTest.php]: `Cannot not` in sprintf

File: tests/Console/Command/ReadmeCommandTest.php (hunk 5 above)
Comment (keradus) [inline tests/Console/Command/ReadmeCommandTest.php]: adverb shall be used

File: tests/Cache/FileHandlerTest.php [hunk 6]
```diff
@@ -103,7 +103,7 @@ public function testWriteThrowsIOExceptionIfFileCanNotBeWritten()
         $file = __DIR__.'/non-existent-directory/.php_cs.cache';
 
         $this->setExpectedExceptionRegExp('Symfony\Component\Filesystem\Exception\IOException', sprintf(
-            '#^Failed to write file "%s"(, ".*")?.#',
+            '#^Failed to write file "%s"(, ".*")?\.#',
```
Comment (keradus) [inline tests/Cache/FileHandlerTest.php]: nice catch

File: tests/Cache/FileHandlerTest.php [hunk 7]
```diff
@@ -103,7 +103,7 @@ public function testWriteThrowsIOExceptionIfFileCanNotBeWritten()
...
             preg_quote($file)
```
Comment (keradus) [inline tests/Cache/FileHandlerTest.php]: could you add here `, '#'` please ?

File: tests/Console/Command/ReadmeCommandTest.php (hunk 5 above)
Comment (SpacePossum) [inline tests/Console/Command/ReadmeCommandTest.php]: which is?

Comment (keradus) [inline tests/Console/Command/ReadmeCommandTest.php]: http://wikidiff.com/successfully/successful

File: tests/Console/Command/ReadmeCommandTest.php (hunk 1 above)
Comment (SpacePossum) [inline tests/Console/Command/ReadmeCommandTest.php]: done

File: tests/Console/Command/ReadmeCommandTest.php (hunk 2 above)
Comment (SpacePossum) [inline tests/Console/Command/ReadmeCommandTest.php]: done

File: tests/Console/Command/ReadmeCommandTest.php (hunk 3 above)
Comment (SpacePossum) [inline tests/Console/Command/ReadmeCommandTest.php]: well PR against it, I won't change it

File: tests/Console/Command/ReadmeCommandTest.php (hunk 4 above)
Comment (SpacePossum) [inline tests/Console/Command/ReadmeCommandTest.php]: tested again, cannot be removed, won't change

File: tests/Console/Command/ReadmeCommandTest.php (hunk 5 above)
Comment (SpacePossum) [inline tests/Console/Command/ReadmeCommandTest.php]: done

File: tests/Console/Command/ReadmeCommandTest.php (hunk 4 above)
Comment (julienfalque) [inline tests/Console/Command/ReadmeCommandTest.php]: 👍 I guess I confused with `CommandTester`

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request closed without merging**